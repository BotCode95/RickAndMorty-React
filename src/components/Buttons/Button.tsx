
interface Props {
    className : string,
    onClick? : () => void | JSX.Element,
    children: string | JSX.Element,
    type? : 'submit' | 'button' | 'reset'
}
export const Button = ({className,onClick, children, type = "button"} : Props) => {
  return (
      <button className={className} onClick={onClick} type={type}>
          {children}
      </button>
  );
};

