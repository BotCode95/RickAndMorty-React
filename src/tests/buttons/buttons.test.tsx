import React from 'react'
import {shallow} from 'enzyme'
import {Button} from '../../components/Buttons/Button'

describe(('Test in <Buttons/>'), ()=> {

    const className: string = "btn btn-danger col-2 ms-2 mb-5";
    const children : string = "Next Page";
    
    test('showed render button next', () => {
        const wrapper = shallow( 
            <Button 
                className={className}
            >{children}
            </Button> );

        expect(wrapper).toMatchSnapshot();
    });


    test('increment page', () => {
        let page = 1;
        const onClick = () => {page++};
         const wrapper = shallow( 
            <Button 
                className={className}
                onClick={onClick}
            >{children}
            </Button> );

        wrapper.find('button').at(0).simulate('click');
        
    });
})