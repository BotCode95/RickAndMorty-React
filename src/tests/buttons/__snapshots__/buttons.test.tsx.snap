// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Buttons/> showed render button next 1`] = `
<button
  className="btn btn-danger col-2 ms-2 mb-5"
  type="button"
>
  Next Page
</button>
`;
